schemas/v2/schema.yaml

```diff
@@ -43,403 +36,413 @@ components:
-            If specified, the whole parameters object can be referenced as `$parameters`.
-            Scope of reference `$parameters`: the reference is shared among all task roles.
-            Specify name and value of all the referencable parameters that will be used in the whole job template.
-            Can be referenced by `<% $parameters.param1 %>`, `<% $parameters.param2 %>`.
-          additionalProperties: true
+            protocolVersion: 2
+            name: tf_example
+            type: dockerimage
+            version: latest
+            contributor: Alice
+            description: python3.5, tensorflow
+            auth:
+              username: user
+              password: <% $secrets.docker_password %>
+              registryuri: openpai.azurecr.io
+            uri: openpai/pai.example.tensorflow
+    ScriptOrOutputComponent:
+      allOf:
+        - $ref: '#/components/schemas/OpenPAIComponent'
+        - type: object
+          properties:
+            type:
+              type: string
+              description: Component type.
+              enum:
+                - script
+                - output
+            uri:
+              type: string
+          required:
+            - name
+            - type
+            - uri
           example:
-            model: resnet20
-            batchsize: 32
-        secrets:
-          type: object
-          description: |
-            If sensitive information including password or API key is needed in the protocol,
-            it should be specified here in secrets section and referenced as `$secrets`.
-            Scope of reference `$secrets`: the reference is shared among all task roles and docker image's `auth` field.
-            A system that supports PAI protocol should keep the secret information away from
-            unauthorized users (how to define unauthorized user is out of the scope of this protocol).
-            For example, the yaml file used for job cloning, the stdout/stderr should protect all information marked as secrets.
-            Specify name and value of all secrets that will be used in the whole job template.
-            Can be referenced by `<% $secrets.secret1 %>`, `<% $secrets.secret2 %>`.
-          additionalProperties: true
+            output_example:
+              value:
+                protocolVersion: 2
+                name: tensorflow_cifar10_model
+                type: output
+                version: latest
+                contributor: Alice
+                description: cifar10 data output
+                uri: hdfs://10.151.40.179:9000/core/cifar10_model
+            script_example:
+              value:
+                protocolVersion: 2
+                name: tensorflow_cnnbenchmarks
+                type: script
+                version: 84820935288cab696c9c2ac409cbd46a1f24723d
+                contributor: MaggieQi
+                description: tensorflow benchmarks
+                uri: github.com/MaggieQi/benchmarks
```

Comment:
the examples are not correct, pls use the original array example in `prerequisites` instead

Reply:
move the Prerequisites example back to `prerequisites` in `JobProtocol` other than in separate Prerequisite